openvpn/client/cliproto.hpp

```diff
@@ -443,12 +443,27 @@ class Session : ProtoContextCallbackInterface,
             if (buf.size())
             {
                 const ProtoContext::ProtoConfig &c = proto_context.conf();
+
+                const uint8_t* packet_data = buf.data();
+                bool df = true;
+
+                // Check if the packet is IPv4
+                if (IPCommon::version(packet_data[0]) == IPCommon::IPv4 && buf.size() >= sizeof(struct IPv4Header))
+                {
+                    // The Flags field is in the 6th byte (starting from index 6) of the IPv4 header
+                    uint16_t flags_and_fragment_offset = ntohs(*(uint16_t*)&packet_data[6]);
```

Comment:
Since we already have `IPv4Header` struct with all the fields, you might want to add DF bit mask there and reuse it.